draft/phoenix_presence_with_live_view.md

```diff
@@ -0,0 +1,146 @@
+
+Let's see how that would work.
+
+## Who's reading?
+
+We are going to build a very simple but powerful feature for the blog you are currently reading. We will add an indicator at the top of the page that shows how many people are currently reading this page. So where do we want this indicator? Preferable this would reside on the same page as a blog post so that it would indicate for every individual blog post how many users are currently reading that article. To do this, we'll create a very small LiveView component inserted in the markdown of our post. Just like we did in my [last post](https://realworldphoenix.com/blog/2020-01-28/rendering_markdown) when I rendered scrambled text. At a later stage I will probably want to make it more generic, but for now let's focus on getting this to work for this blog post.
+
+If you head over to my [interactive blog](https://realworldphoenix.com/blog/2020-02-11/simple_phoenix_presence), you can see this counter in action!
+
+This is how we embed the LiveView Component into out markdown post:
+
+```elixir
+<%= Phoenix.LiveView.Helpers.live_render(@conn, RealWorldPhoenixWeb.Live.ReaderCount, session: %{slug: "simple_phoenix_presence"}) %>
+```
+
+## LiveView or LiveView Component?
+
+LiveView is maturing quickly and currently also has a concept of Components. Which basically are small building blocks that either are stateful or stateless. I am definitely planning on doing a writeup on using LiveView components, but for our usecase we simply need the basic LiveView.
+
+Let's create a basic LiveView boilerplate:
+
+```elixir
+defmodule RealWorldPhoenixWeb.Live.ReaderCount do
+  use Phoenix.LiveView
+
+  @moduledoc """
+    A small LiveView that shows the number of readers of a post using Phoenix Presence
+  """
+
+  def render(assigns) do
+    ~L"""
+      Readers: <\%= @reader_count %>
+    """
+  end
+
+  def mount(_session, socket) do
+    {:ok, assign(socket, :reader_count, 0)}
+  end
+end
+```
+
+Ok, so now we need to hook up Presence. Let's see how that would work. First we'll head over to the [docs](https://hexdocs.pm/phoenix/Phoenix.Presence.html) and see how it is setup. The first thing I notice is this: _Provides Presence tracking to processes and channels._. We are not really using Channels, but our LiveView is a process so I guess that should work, right?
+
+## Presence Setup
+
+Let's first setup Presence using the guide provides by the Phoenix Team:
+
+### 1. Setup a Presence Module in our app:
+
+```elixir
+defmodule RealWorldPhoenix.Presence do
+  use Phoenix.Presence, otp_app: :real_world_phoenix,
+                        pubsub_server: MyApp.PubSub
```

Comment:
shouldnt this be `RealWorldPhoenix.PubSub` ?

Reply:
good catch ⚾️